### PR TITLE
Refine Error Message for New Data API

### PR DESCRIPTION
**Pull request by zhhsplendid**
The shape doesn't match error will be like:

_"The Variable u'data' should have dimensions = 4, shape = (-1, 3, 4, 8), but received feeded shape [10, 3, 4, 5]"_

The dtype doesn't match error will be like:

_"The data type of feeded Variable u'label' must be 'int64', but received 'float64'"_

### CODE REVIEW

File: python/paddle/fluid/executor.py [hunk 1]
```diff
@@ -204,13 +204,14 @@ def check_feed_shape_type(var, feed):
     """
     if var.desc.need_check_feed():
         if not dimension_is_compatible_with(feed.shape(), var.shape):
-            raise ValueError('Cannot feed value of shape %r for Variable %r, '
-                             'which has shape %r' %
-                             (feed.shape, var.name, var.shape))
+            raise ValueError(
+                'ShapeError: The feeded Variable %r should have dimensions = '
```
Comment (chenwhql) [inline python/paddle/fluid/executor.py]: 后来补充了一条规则，建议把这里的`ShapeError`去掉

image

File: python/paddle/fluid/executor.py [hunk 2]
```diff
@@ -204,13 +204,14 @@ def check_feed_shape_type(var, feed):
...
+                '%d, shape = %r, but received feeded shape %r' %
+                (var.name, len(var.shape), var.shape, feed.shape()))
         if not dtype_is_compatible_with(feed._dtype(), var.dtype):
-            raise ValueError('Cannot feed value of type %r for Variable %r, '
-                             'which has type %r' %
-                             (feed._dtype(), var.name, var.dtype))
+            raise ValueError(
+                'The data type of feeded Variable %r must be %r, but received %r'
+                % (var.name, var.dtype, feed._dtype()))
```
Comment (chenwhql) [inline python/paddle/fluid/executor.py]: 这里是否有必要对用户显示numpy的类型？ @luotao1 ，用户输入的`float32`这里会打印`VarType.FP32`，是否会不太友好？

Comment (luotao1) [reply python/paddle/fluid/executor.py]: 打印VarType.FP32肯定不是很友好，这里可以用示例中的`convert_dtype`来转换下么？

Comment (zhhsplendid) [reply python/paddle/fluid/executor.py]: Done

File: python/paddle/fluid/executor.py (hunk 1 above)
Comment (zhhsplendid) [reply python/paddle/fluid/executor.py]: Done

**Review by luotao1 — APPROVED** [review]
LGTM

**Pull request merged**